### PR TITLE
Feat creating day all page and getting comment data in detail page

### PR DESCRIPTION
**Pull request by yuki-o1o5**
## Proposed Changes
Update main/index.jsx and main/[id].jsx
Update All/index.jsx and All/[id].jsx

#### Code changes and Reasons
- Add a link in order to go to main/[id] page from main page
- Change Article number in main page in order to increase the number automatically  
- Add comment out all components in order to let GitHub know differences between uppercase and lowercase 
- Add populateGiftConfigurationParameters function to populate data
- Create some components in order to use all page and all/[id] page
- Create all page and all/[id] page and refactor them in order to use getStaticProps and getStaticPath

#### Related Issues
- #4 
- #8 
- #11 
- #13 
- #14  
- #16 

## Screen Cast
https://user-images.githubusercontent.com/108368307/207995264-9f91c733-894d-4faf-acb6-f0536bfbaa18.mp4

### CODE REVIEW

File: pages/all/[id].jsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    );
+    const eachStoryDetails = await detail.json();
+    return eachStoryDetails;
+  };
+  const story = await getDetailUrl(storyId);
+
+  // 3.This is the top comment.
+  const getCommentUrl = async (commentId) => {
+    const res = await fetch(
+      "https://hacker-news.firebaseio.com/v0/item/" +
```
Comment (yuyaohshimo) [inline pages/all/[id].jsx:26]: minor: You should use template literals because it is easy to understand.

File: pages/all/[id].jsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    );
+    const comments = await res.json();
+    return comments;
+  };
+
+  // 3.This is the comments of the top comment .
+  const topComment = await getCommentUrl(story.kids[0]);
+  // console.log("topComment", topComment);
+
+  const topCommentReplies = await Promise.all(
```
Comment (yuyaohshimo) [inline pages/all/[id].jsx:38]: You should add null checker for `topComment.kids` before running `Promise.all`.

**Review by yuyaohshimo — CHANGES_REQUESTED** [review]
Can you add a screencast so that I can check the screens and behavior?

File: pages/all/[id].jsx (hunk 1 above)
Comment (yuki-o1o5) [reply pages/all/[id].jsx]: I understand. I changed it to template literals.
 --> [link1](https://github.com/yuki-o1o5/Hacker-News-in-Japanese/commit/3876ce96a795528ac285f8935e62166b9f8b76bd)
 --> [link2](https://github.com/yuki-o1o5/Hacker-News-in-Japanese/commit/624266427ed7ac66bb6b70c2c76a408a8476eb15)

Comment (yuyaohshimo) [reply pages/all/[id].jsx]: GJ!

File: pages/all/[id].jsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  const getCommentUrl = async (commentId) => {
+    const res = await fetch(
+      `https://hacker-news.firebaseio.com/v0/item/${commentId}.json?print=pretty`
+    );
+    const comments = await res.json();
+    return comments;
+  };
+
+  // 3.This is the comments of the top comment .
+  const topComment = story.kids ? await getCommentUrl(story.kids[0]) : "";
```
Comment (yuyaohshimo) [inline pages/all/[id].jsx:33]: I think the following code is easy to understand more to check `undefined`:

```javascript

const topComment = story.kids ? await getCommentUrl(story.kids[0]) : "";

let topCommentReplies = [];
if (!topComment.kids) {
  topCommentReplies = await Promise.all(
    topComment.kids.map((topCommentKid) => getCommentUrl(topCommentKid))
  );
}
```

Comment (yuki-o1o5) [reply pages/all/[id].jsx:33]: I understand. I changed the upper code. --> [link](https://github.com/yuki-o1o5/Hacker-News-in-Japanese/commit/29822635d7e040767a2da2534576f8bc79d9f399)

**Pull request merged**